src/dialogs/voice-assistant-setup/cloud/cloud-step-signup.ts

```diff
@@ -0,0 +1,212 @@
+                minlength="8"
+                required
+                @keydown=${this._keyDown}
+                validationMessage=${this.hass.localize(
+                  "ui.panel.config.cloud.register.password_error_msg"
+                )}
+              ></ha-password-field>`}
+      </div>
+      <div class="footer side-by-side">
+        ${this._state === "VERIFY"
+          ? html`<ha-button
+                @click=${this._handleResendVerifyEmail}
+                .disabled=${this._requestInProgress}
+                >Send the email again</ha-button
+              ><ha-button
+                unelevated
+                @click=${this._login}
+                .disabled=${this._requestInProgress}
+                >I clicked the link</ha-button
+              >`
+          : html`<ha-button
+                @click=${this._signIn}
+                .disabled=${this._requestInProgress}
+                >Sign in</ha-button
+              >
+              <ha-button
+                unelevated
+                @click=${this._handleRegister}
+                .disabled=${this._requestInProgress}
+                >Next</ha-button
+              >`}
+      </div>`;
+  }
+
+  private _signIn() {
+    fireEvent(this, "cloud-step", { step: "SIGNIN" });
+  }
+
+  private _keyDown(ev: KeyboardEvent) {
+    if (ev.key === "Enter") {
+      this._handleRegister();
+    }
+  }
+
+  private async _handleRegister() {
+    const emailField = this._emailField;
+    const passwordField = this._passwordField;
+
+    if (!emailField.reportValidity()) {
+      passwordField.reportValidity();
+      emailField.focus();
+      return;
+    }
+
+    if (!passwordField.reportValidity()) {
+      passwordField.focus();
+      return;
+    }
+
+    const email = emailField.value.toLowerCase();
+    const password = passwordField.value;
+
+    this._requestInProgress = true;
+
+    try {
+      await cloudRegister(this.hass, email, password);
+      this._email = email;
+      this._password = password;
+      this._verificationEmailSent();
+    } catch (err: any) {
+      this._password = "";
+      this._error =
+        err && err.body && err.body.message
+          ? err.body.message
+          : "Unknown error";
+    } finally {
+      this._requestInProgress = false;
+    }
+  }
+
+  private async _handleResendVerifyEmail() {
+    if (!this._email) {
+      return;
+    }
+    try {
+      await cloudResendVerification(this.hass, this._email);
+      this._verificationEmailSent();
+    } catch (err: any) {
+      this._error =
+        err && err.body && err.body.message
+          ? err.body.message
+          : "Unknown error";
+    }
+  }
+
+  private _verificationEmailSent() {
+    this._state = "VERIFY";
+
+    setTimeout(() => this._login(), 5000);
+  }
+
+  private async _login() {
+    if (!this._email || !this._password) {
+      return;
+    }
+
+    try {
+      await cloudLogin(this.hass, this._email, this._password);
```

Comment:
Bit weird that we have login code in 2 of the steps but it making it DRY may complicate the code too much